stringer.go

```diff
@@ -51,9 +56,16 @@ func (n *node) String() string {
 }
 
 func (k key) String() string {
+	// ~tally.Scope means optional
+	// ~tally.Scope:foo means named optional
+	// [io.Reader]:foo refers to a group of io.Readers called 'foo'
```

Comment:
I suppose this is okay.

Would this be clearer with different notation?

```
tally.Scope
tally.Scope[optional]
tally.Scope[optional, name=foo]
tally.Scope[group=foo]
```

If we add a new tag, for example, I'm not sure how we'd extend the current system.

Reply:
I like this. Will change in a follow up.

Reply:
#168